File: abc/abc283/b.cpp
// https://atcoder.jp/contests/abc283/submissions/37485703

#include <bits/stdc++.h>
using namespace std;

int main() {
  cin.tie(0);
  ios_base::sync_with_stdio(0);

  int n;
  cin >> n;

  vector<int> a(n);
  for(auto &z : a) cin >> z;

  int q;
  cin >> q;
  while(cin >> q) {
    if(q == 1) {
      int k, x;
      cin >> k >> x;
      k--;
      a[k] = x;
    } else {
      int k;
      cin >> k;
      k--;
      cout << a[k] << endl;
    }
  }
}

File: abc/abc283/b.py
# https://atcoder.jp/contests/abc283/submissions/37515204

n = int(input())
a = list(map(int, input().split()))

q = int(input())
for _ in range(q):
  query = list(map(int, input().split()))
  query[1] -= 1

  if query[0] == 1:
    a[query[1]] = query[2]
  
  if query[0] == 2:
    print(a[query[1]])
